catkit/hardware/iris_ao/iris_ao_controller.py

```diff
@@ -85,10 +85,13 @@ def _communicate(self, input):
             self.instrument = None  # If it died, there's nothing to close, so don't.
             raise RuntimeError(f"{self.config_id} unexpectedly closed.") from error
 
```

Comment:
A sleep in here would make this more robust and likely for the poll to catch the terminated process, however, it would be at the obvious cost of slowing things down. It would prob have to be the order of 1s, so I don't suggest we do it.

Reply:
@ivalaginja If/when bug-hunting I would add a ``time.sleep(1)`` here (maybe even 2s).

Reply:
Right - I just wouldn't know how to hunt for bugs. The stalling seems to happen in random iterations of experiments that last 2+ hours 🤷

Reply:
You could check the command file, whilst it's still hanging, in case it was sent some bogus data. You could also try connecting to it from the GUI.

Reply:
For the latter, you prob don't want the sleep, such that you can do it whilst it's still hanging.

You could take a look at the cams in sharpcap, might help to confirm that it actually died.

Reply:
Yeah I did check Sharpcap, it was just holding the DH. All DMs seemed to just keep holding their latest shape.